apt/private/apt_deb_repository.bzl

```diff
@@ -122,39 +142,24 @@ def _create(rctx, sources, archs):
             # on misconfigured HTTP servers)
             url = url.rstrip("/")
 
-            rctx.report_progress("Fetching package index: {}/{} for {}".format(dist, comp, arch))
-            (output, _) = _fetch_package_index(rctx, url, dist, comp, arch, "")
-
-            # TODO: this is expensive to perform.
-            rctx.report_progress("Parsing package index: {}/{} for {}".format(dist, comp, arch))
-            _parse_repository(state, rctx.read(output), url)
-
-    return struct(
-        package_versions = lambda **kwargs: _package_versions(state, **kwargs),
-        virtual_packages = lambda **kwargs: _virtual_packages(state, **kwargs),
-        package = lambda **kwargs: _package(state, **kwargs),
-    )
+            index = "{}/{} for {}".format(dist, comp, arch)
 
-deb_repository = struct(
-    new = _create,
-)
+            rctx.report_progress("Fetching package index: %s" % index)
+            output = _fetch_package_index(rctx, url, dist, comp, arch)
 
-# TESTONLY: DO NOT DEPEND ON THIS
-def _create_test_only():
-    state = struct(
-        packages = dict(),
-        virtual_packages = dict(),
-    )
+            rctx.report_progress("Parsing package index: %s" % index)
+            _parse_package_index(state, output, url)
 
     return struct(
-        package_versions = lambda **kwargs: _package_versions(state, **kwargs),
-        virtual_packages = lambda **kwargs: _virtual_packages(state, **kwargs),
-        package = lambda **kwargs: _package(state, **kwargs),
-        parse_repository = lambda contents: _parse_repository(state, contents, "http://nowhere"),
-        packages = state.packages,
-        reset = lambda: state.packages.clear(),
+        package_versions = lambda arch, name: state.packages.get((arch, name), {}).keys(),
+        virtual_packages = lambda arch, name: state.virtual_packages.get((arch, name), []),
+        package = lambda arch, name, version: state.packages.get((arch, name, version)),
     )
 
-DO_NOT_DEPEND_ON_THIS_TEST_ONLY = struct(
```

Comment:
Again, i don't see why this needs to be change, it's already working and there is no value in changing things without a reason. If this change is needed in a larger context, eg if  your stack has a fix for something that needs this change, please put it right next to that fix to help me reason why this needs to change.

Reply:
This was in the same spirit as the test helpers that I added, standardizing on having a `_TEST_SUITE_PREFIX` constant, etc.

Basically, it was meant to be a clean way to link the stuff that's not publicly accessible for testing purposes only. This way, we always import the module in the test and test the public parts of the struct and if there are parts that are private and we want to test, I thought it would be easier to have a `__test__` sub-struct in-place and not have to create another struct, with a long name in there (the `__test__` already does signal that it's for testing only, I think).

I'll go back to the `DO_NOT_DEPEND_ON_THIS_TEST_ONLY`...

Reply:
@thesayyn I'm really looking into this, I was going to make the change but here's the thing. As mentioned, once I've added proper mocking, the tests don't need any of the `_create_test_only()` functionality. That is, all of this functionality is superseded by the mocks, which is the "added value".

So, TL;DR added value == better testing, like with the helper stuff you like in the tests. Is that OK? 🙏